### DISCUSSION TITLE
net rate must be in bits per second ?

### DISCUSSION THREAD

**Discussion by alexwbaule** [mathoudebine/turing-smart-screen-python#259] [Ideas]
Normally, today using tools to mesure the net speed, we see the speed returned in bits per second.
1 MByte = 8 Mbit 

What do you think the idea to update the mesurements to be in (UNIT)bps ? 

What now is M/s K/s will turn in Mb/s Kb/s  (or Mbps, Kbps)

Looking to the code is easy to convert to bits (multiply by 8) and some adapts to bytes2human , to reflect the correct symbol...

**Comment by joevoisin** [comment]
I looked for a few min and found away to do this if you are still looking.

in sensors_librehardwaremonitor.py you can just modify what it returns and add a x8 to both the rate variables.  (saves me from seeing where else it is used.. just change it right at the source.
look for `class Net(sensors.Net):` and change the last line 
        `return upload_rate*8, uploaded, download_rate*8, downloaded`

to clean up the B/s to make it show Kbps/Mbps/Gbps instead just make the change beliw.  side effect is that it may display Bbps.. but find that doesn't relaly matter much. my system is never idle enough for that to matter.  

in stats.py find 
`value=f"{bytes2human(rate, '%(value).1f %(symbol)s/s')}",`
and change it to
`value=f"{bytes2human(rate, '%(value).1f %(symbol)sbps')}",`

I am not a python expert by any stretch of the imagination.. but came across this when I was googling to see if anyone else had already changed this.. turns out it's pretty simple.. but it won't survive upgrades to the code from the author. be nice if it would return both.. let us chose which one to display.. displaying Bytes/s make sense with disk.. but not network..
